app/components/scroll-to-top.tsx

```diff
@@ -0,0 +1,47 @@
+"use client";
+
+import {useEffect} from "react";
+import {useAnimationControls, useScroll, motion, Variants} from "framer-motion";
+import {FaArrowUp} from "react-icons/fa"
+
+const isBrowser = () => typeof window != 'undefined';
+
+const ScrollToTopContainerVariants: Variants = {
+    hide: {opacity: 0, y: 100},
+    show: {opacity: 1, y: 0},
+}
+
+const scrollToTop = () => {
+    if (!isBrowser()) {
+        return;
+    }
+    window.scrollTo({top: 0, behavior: 'smooth'});
+};
+
+const ScrollToTopButton = () => {
+    const {scrollYProgress} = useScroll();
+    const controls = useAnimationControls();
+
+    useEffect(() => {
+        return scrollYProgress.on('change', (latestValue) => {
+            if (latestValue > 0.5) {
```

Comment:
I think it is better to use `scrollY` which uses the absolute y-value instead of `scrollYProgress` which uses relative value. Intuitively, we want the scroll-to-top button to appear when the user scrolls far enough from the top of the page, and this should be irrespective of the page content itself.

Reply:
For the offset, maybe `window.innerHeight` is ok.